### PR TITLE
[SYCLomatic] Refine bindless image helper functions.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by wenhuiNi — APPROVED** [review]

**Review by ziranzha — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 1]
```diff
@@ -228,12 +229,13 @@ dpct_memcpy(void *src, sycl::ext::oneapi::experimental::image_mem_handle dest,
   const auto dest_offset =
       sycl::range<3>(w_offset_dest / ele_size, h_offset_dest, 0);
   const auto copy_extend = sycl::range<3>(w / ele_size, h, 0);
-  return q.ext_oneapi_copy(src, src_offset, src_extend, dest, dest_offset,
-                           desc_dest, copy_extend);
+  return q.ext_oneapi_copy(const_cast<void *>(src), src_offset, src_extend,
```
Comment (ziranzha) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp:233]: It's better to ask CodePlay to refine the API interface.

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 2]
```diff
@@ -248,9 +250,9 @@ dpct_memcpy(void *src, sycl::ext::oneapi::experimental::image_mem_handle dest,
         sycl::range<3>(w_offset_dest / ele_size, h_offset_dest, 0);
     const auto copy_extend =
         sycl::range<3>((w - w_offset_dest) / ele_size, 1, 0);
-    event_list.push_back(q.ext_oneapi_copy(src, src_offset, src_extend, dest,
-                                           dest_offset, desc_dest,
-                                           copy_extend));
+    event_list.push_back(q.ext_oneapi_copy(const_cast<void *>(src), src_offset,
```
Comment (ziranzha) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp:255]: Same as above.

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 3]
```diff
@@ -261,8 +263,9 @@ dpct_memcpy(void *src, sycl::ext::oneapi::experimental::image_mem_handle dest,
       sycl::range<3>(w_offset_dest / ele_size, h_offset_dest, 0);
   const auto copy_extend =
       sycl::range<3>((s - offset_src - w_offset_dest) / ele_size, 1, 0);
-  event_list.push_back(q.ext_oneapi_copy(src, src_offset, src_extend, dest,
-                                         dest_offset, desc_dest, copy_extend));
+  event_list.push_back(q.ext_oneapi_copy(const_cast<void *>(src), src_offset,
```
Comment (ziranzha) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp:269]: Same as above

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 1 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp:233]: Add TODO.

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 2 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp:255]: Add TODO.

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 3 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp:269]: Add TODO.

**Review by ziranzha — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**